### PR TITLE
[REF] toward dev/core#1948 - Move otherRelationships calculation to its own function

### PR DESCRIPTION
**Pull request by demeritcowboy**
Overview
----------------------------------------
Step toward https://lab.civicrm.org/dev/core/-/issues/1948

This just copy/pastes a block into its own function, partly to make it easier to test and partly to isolate for fixing the problem in the ticket.

Before
----------------------------------------

After
----------------------------------------

Technical Details
----------------------------------------

Comments
----------------------------------------
Has test

### CODE REVIEW

File: CRM/Case/XMLProcessor/Report.php [hunk 1]
```diff
@@ -880,44 +880,7 @@ public static function printCaseReport() {
...
-            array($r['email'] => 'email_' . rand(10000, 100000))
-          );
-        }
-        $r['email'] = $report->redact($r['email'], TRUE, $report->_redactionStringRules);
-      }
-      if (!array_key_exists($r['id'], $caseRelationships)) {
-        $otherRelationships[] = $r;
-      }
-    }
+    $otherRelationships = $this->getOtherRelationships($clientID, $caseID, $isRedact, $report, $caseRelationships);
```
Comment (eileenmcnaughton) [inline CRM/Case/XMLProcessor/Report.php:883]: I think this needs to be a static function - my ide is complaining it is not in object context

File: CRM/Case/XMLProcessor/Report.php [hunk 2]
```diff
@@ -986,4 +949,58 @@ public static function printCaseReport() {
...
+   *
+   * @param int $clientID
+   * @param int $caseID
+   * @param bool $isRedact
+   * @param CRM_Case_XMLProcessor_Report $report
+   * @param array $caseRelationships
+   *
+   * @return array
+   */
+  protected function getOtherRelationships(int $clientID, int $caseID, bool $isRedact, CRM_Case_XMLProcessor_Report $report, array $caseRelationships): array {
```
Comment (eileenmcnaughton) [inline CRM/Case/XMLProcessor/Report.php:963]: I don't think caseID is used?

**Comment by eileenmcnaughton** [comment]
I tried doing this extraction and got a similar result with the 2 differences I commented on

File: CRM/Case/XMLProcessor/Report.php (hunk 2 above)
Comment (demeritcowboy) [reply CRM/Case/XMLProcessor/Report.php:963]: Thanks for looking. Ah I forgot to remove it. I did the fix, which does use caseID, then went back and tried to make a smaller PR but forgot to take this out. `$rindex` too.

File: CRM/Case/XMLProcessor/Report.php (hunk 1 above)
Comment (demeritcowboy) [reply CRM/Case/XMLProcessor/Report.php:883]: I guess it doesn't come up in running since it's never called statically. I might rethink this a little.

**Pull request closed without merging**